File: LeetCode/src/13.rs
struct Solution;

impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        let mut result = 0;
        let mut number: i32;
        for char in s.chars().rev() {
            match char {
                'I' => number = 1,
                'V' => number = 5,
                'X' => number = 10,
                'L' => number = 50,
                'C' => number = 100,
                'D' => number = 500,
                'M' => number = 1000,
                _ => panic!("Invalid roman number"),
            }
            if number * 4 < result {
                result -= number;
            } else {
                result += number;
            }
        }
        result
    }
}

File: Go/13.go
package main

func romanToInt(s string) int {
	reverseString := func(str *string) {
		result := ""
		for _, char := range *str {
			result = string(char) + result
		}
		*str = result
	}
	reverseString(&s)

	result := 0
	var number int

	for _, char := range s {
		switch char {
		case 'I':
			number = 1
		case 'V':
			number = 5
		case 'X':
			number = 10
		case 'L':
			number = 50
		case 'C':
			number = 100
		case 'D':
			number = 500
		case 'M':
			number = 1000
		default:
			panic("Invalid roman number")
		}

		if number*4 < result {
			result -= number
		} else {
			result += number
		}
	}
	return result
}
